executor/insert_test.go

```diff
@@ -398,6 +398,14 @@ func TestInsertWrongValueForField(t *testing.T) {
 	tk.MustExec(`SET @@sql_mode='STRICT_TRANS_TABLES'`)
 	tk.MustGetErrMsg(`INSERT INTO ts (id, time1) VALUES (2, TIMESTAMP '1018-12-24 00:00:00')`, `[table:1292]Incorrect timestamp value: '1018-12-24 00:00:00' for column 'time1' at row 1`)
 	tk.MustExec(`DROP TABLE ts`)
+
+	tk.MustExec(`CREATE TABLE t0(c0 SMALLINT AUTO_INCREMENT PRIMARY KEY);`)
+	tk.MustExec(`INSERT IGNORE  INTO t0(c0) VALUES (194626268);`)
+	tk.MustExec(`INSERT IGNORE  INTO t0(c0) VALUES ('*')`)
+	tk.MustQuery(`SHOW WARNINGS`).Check(testkit.Rows(
+		`Warning 1366 Incorrect smallint value: '*' for column 'c0' at row 1`,
+		`Warning 1690 constant 32768 overflows smallint`,
```

Comment:
seems the warning message indicates the internal behaviour is not same as MySQL, MySQL has warning of

```
MySQL [test]> show warnings;
+---------+------+-------------------------------------------------------+
| Level   | Code | Message                                               |
+---------+------+-------------------------------------------------------+
| Warning | 1366 | Incorrect integer value: '*' for column 'c0' at row 1 |
| Warning | 1062 | Duplicate entry '32767' for key 'PRIMARY'             |
+---------+------+-------------------------------------------------------+
```

So we can find another case that TiDB is different with MySQL

```
MySQL [test]> alter table t0 add column c1 int default 1;
Query OK, 0 rows affected (0.081 sec)
Records: 0  Duplicates: 0  Warnings: 0

MySQL [test]> INSERT ignore INTO t0(c0) VALUES ('*') on duplicate key update c1 = 2;
Query OK, 2 rows affected, 1 warning (0.040 sec)

MySQL [test]> select * from t0;
+-------+------+
| c0    | c1   |
+-------+------+
| 32767 |    2 |
+-------+------+
1 row in set (0.001 sec)
```

Reply:
It is another bug. 

Sad.

Reply:
I found this behaviour in doc

> When an AUTO_INCREMENT integer column runs out of values, a subsequent INSERT operation returns a duplicate-key error. This is general MySQL behavior.

https://dev.mysql.com/doc/refman/8.0/en/innodb-auto-increment-handling.html

Reply:
OK